include/response_reader.rb

```diff
@@ -105,10 +105,8 @@ def url_and_args_from_location
     url << "?#{uri.query}" if uri.query.to_s.length > 0
     url = url.gsub(/\A\//, "")
 
-    args = {host: uri.host}
+    args = @http2.args.merge(host: uri.host)
```

Comment:
Might be possible though possible more theoretically due to the url from the uri itself. But better safe than sorry.

Fixed in f520422